semgrep-core/tests/Test.ml

```diff
@@ -52,6 +52,7 @@ let tests () = List.flatten [
   Unit_memory_limit.tests;
   Unit_SPcre.tests;
   Unit_regexp_engine.tests;
+  Unit_immutable_buffer.tests;
```

Comment:
Are there any integration tests to show this working? Or is that waiting on the autofix test suite? (I saw some discussion but didn't pay attention)

Reply:
In the summary, I wrote:

> I'm comfortable with the existing tests for this change, but as we
improve autofix I think we'll want some better testing infrastructure.
The existing end-to-end tests are thorough, but they are a bit clunky.
It would be nice to have semgrep-core tests that verify that AST-based
autofix is working correctly.

Does that answer your question?